components/mpas-ocean/src/shared/mpas_ocn_time_average_coupled.F

```diff
@@ -584,4 +549,129 @@ subroutine ocn_time_average_coupled_accumulate(statePool, forcingPool, timeLevel
+        real (kind=RKIND), dimension(:), pointer :: ssh
+
+        type (mpas_pool_type), pointer :: tracersPool
+
+        real (kind=RKIND), dimension(:,:,:), pointer :: activeTracers
+        integer, pointer :: indexTemperature, indexSalinity, nGlcZLevels, &
+                            nCells, nAccumulatedCoupled
+        integer :: iLevel, iLevelGlc, iLevelAbove, iLevelBelow
+
+        real (kind=RKIND) :: freezingTemp, glcz, zMin, zMax, zAbove, zBelow, &
+                             alpha, salin, temp, press
+
+        call mpas_pool_get_dimension(forcingPool, 'nCells', nCells)
+        call mpas_pool_get_dimension(forcingPool, 'nGlcZLevels', nGlcZLevels)
+
+        call mpas_pool_get_array(forcingPool, 'nAccumulatedCoupled', nAccumulatedCoupled)
+
+        call mpas_pool_get_array(forcingPool, 'glcZLevels', glcZLevels)
+        call mpas_pool_get_array(statePool, 'ssh', ssh, timeLevel)
+
+        call mpas_pool_get_array(forcingPool, 'avgThermalForcingAtZLevels', avgThermalForcingAtZLevels)
+        call mpas_pool_get_array(forcingPool, 'avgThermalForcingAtZLevelsMask', avgThermalForcingAtZLevelsMask)
+        call mpas_pool_get_subpool(statePool, 'tracers', tracersPool)
+        call mpas_pool_get_array(tracersPool, 'activeTracers', activeTracers, timeLevel)
+        call mpas_pool_get_dimension(tracersPool, 'index_temperature', indexTemperature)
+        call mpas_pool_get_dimension(tracersPool, 'index_salinity', indexSalinity)
+
+        !$omp parallel
+        !$omp do schedule(runtime)
+        do iLevelGlc = 1, nGlcZLevels
+            ! calculate thermal forcing at identified level for each cell
+            glcz = glcZLevels(iLevelGlc)
+
+            do iCell = 1, nCells
+
+                if (glcz > ssh(iCell) .or. glcz < -bottomDepth(iCell)) then
+                    ! mask out glc levels that are too shallow or too deep
+                    avgThermalForcingAtZLevels(iLevelGlc, iCell) = &
+                        ( avgThermalForcingAtZLevels(iLevelGlc, iCell) * nAccumulatedCoupled &
+                          + 0.0_RKIND) / (nAccumulatedCoupled + 1)
+                    avgThermalForcingAtZLevelsMask(iLevelGlc, iCell) = &
+                        ( avgThermalForcingAtZLevelsMask(iLevelGlc, iCell) * nAccumulatedCoupled &
+                          + 0.0_RKIND) / ( nAccumulatedCoupled + 1)
+                    cycle
+                end if
+
+                ! find which zMid values glcz lies between and the linear
+                ! interpolation factor alpha between them
+
+                zMin = zMid(minLevelCell(iCell), iCell)
+                zMax = zMid(maxLevelCell(iCell), iCell)
+                if (glcz > zMin) then
+                    ! above the middle of the top level so no interpolation
+                    iLevelAbove = minLevelCell(iCell)
+                    iLevelBelow = minLevelCell(iCell)
+                    alpha = 0.0_RKIND
+                else if (glcz > zMax) then
```

Comment:
An obvious bug!
```suggestion
                else if (glcz < zMax) then
```

Reply:
Still getting division by zero even with this fix, so investigation is ongoing...